homeassistant/components/volvooncall/binary_sensor.py

```diff
@@ -24,16 +24,28 @@ async def async_setup_platform(
 class VolvoSensor(VolvoEntity, BinarySensorEntity):
     """Representation of a Volvo sensor."""
 
+    def __init__(
+        self,
+        coordinator: VolvoUpdateCoordinator,
+        vin: str,
+        component: str,
+        attribute: str,
+        slug_attr: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(vin, component, attribute, slug_attr, coordinator)
+
+        if self.instrument.device_class in DEVICE_CLASSES:
+            self._attr_device_class = self.instrument.device_class
+
     @property
-    def is_on(self):
-        """Return True if the binary sensor is on, but invert for the 'Door lock'."""
+    def is_on(self) -> bool | None:
+        """Fetch from update coordinator."""
         if self.instrument.attr == "is_locked":
             return not self.instrument.is_on
         return self.instrument.is_on
 
-    @property
-    def device_class(self):
-        """Return the class of this sensor, from DEVICE_CLASSES."""
-        if self.instrument.device_class in DEVICE_CLASSES:
-            return self.instrument.device_class
-        return None
+    @callback
+    def _handle_coordinator_update(self) -> None:
```

Comment:
We can remove this method as it's already implemented by the coordinator entity parent.